CIAOS-Rust/client/ciaos/client.py

```diff
@@ -0,0 +1,238 @@
+import sys
+import os
+sys.path.append(os.path.join(os.path.dirname(__file__), 'util'))
+import requests
+from typing import List
+from util.flatbuffer.flatbuffer_handler import create_flatbuffer, parse_flatbuffer
+import config
+from typing import List, Optional
+
+class Ciaos:
+    def __init__(self):
+        """
+        Initialize Ciaos client with configuration parameters.
+        API URL and user ID are imported from config
+
+        config.API_URL = "<server url>"
+        config.user.id = "your_user_id"
+        """
+
+        # Get user ID from config
+        try:
+            self.user_id = config.user_id
+            if not self.user_id:
+                raise ValueError("User ID must be set in config.user.id")
+        except AttributeError:
+            raise ValueError("config.user.id must be defined")
+
+        self.headers = {
+            "User": self.user_id,
+        }
+
+    def put(self, file_path: str, key: Optional[str] = None):
+        """
+        Uploads files to the server with flexible input options.
+
+        Args:
+            key (Optional[str]): Unique key for the upload. If None and file_path provided, 
+                            uses filename as key.
+            file_path : Path to file to upload.reads file data.
+
+        Returns:
+            requests.Response or None: The server's response or None if an error occurs.
+        """
+        try:
+            # Handle file path only case
+            with open(file_path, 'rb') as file:
+                file_data = file.read()
+                data_list = [file_data]
+                    
+                    # If key not provided, use filename from path
+            if key is None:
+                key = os.path.basename(file_path)
+
+
+            # Create and send flatbuffer data
+            flatbuffer_data = create_flatbuffer(data_list)
+            if flatbuffer_data is None:
+                print("Failed to create FlatBuffers data.")
+                return None
+
+            response = requests.post(
+                f"{config.api_url}/put/{key}", 
+                data=flatbuffer_data, 
+                headers=self.headers
+            )
+
+            if response.status_code == 200:
+                print("Upload successful:", response.text)
+            else:
+                print("Error during upload:", response.text)
+            return response
+
+        except requests.RequestException as e:
+            print("HTTPError during upload:", e)
+            return None
+    
+    def put_binary(self, key: str, data_list: List[bytes]):
+        try:
+            flatbuffer_data = create_flatbuffer(data_list)
+            if flatbuffer_data is None:
+                print("Failed to create FlatBuffers data.")
+                return None
+
+
+            response = requests.post(
+                f"{config.api_url}/put/{key}", 
```

Comment:
Same backend service put works for both put and put binary?

Reply:
yep